schoenrock-rossiter_tom_ndvi.py

```diff
@@ -0,0 +1,656 @@
+
+if student_wd_parts[-2:] == ['earth-analytics', 'data']:
+    print("\u2705 Great - it looks like your working directory is set correctly to ~/earth-analytics/data")
+else:
+    print("\u274C Oops, the autograder will not run unless your working directory is set to earth-analytics/data")
+
+
+# # Figure 1: Plot 1 - Mean NDVI For Each Site Across the Year (50 points)
+# 
+# Create a plot of the mean normalized difference vegetation index (NDVI) for the two different sites in the United States across the year: 
+# 
+# * NDVI on the x axis and formatted dates on the y for both NEON sites on one figure/axis object.
+# * Each site should be identified with a different color in the plot and legend.
+# * The final plot **data should be cleaned to remove the influence of clouds**.
+# * Be sure to include appropriate title and axes labels.
+# 
+# Add additional cells as needed for processing data (e.g. defining functions, etc), but be sure to:
+# * follow the instructions in the code cells that have been provided to ensure that you are able to use the sanity check tests that are provided. 
+# * include only the plot code in the cell identified for the final plot code below
+
+# ## Task 1: 
+# 
+# In the cell below, create a single dataframe containing MEAN NDVI, the site name, 
+# and the date of the data for the HARV site 
+# scene `HARV/landsat-crop/LC080130302017031701T1-SC20181023151837`.  The column names for the  final
+# DataFrame should be`mean_ndvi`, and `site`, and the data should be **indexed on the date**. 
+# 
+# Use the functions that we reviewed in class (or create your own versions of them) to implement your code
+# 
+# ### In the Cell below Place  All Functions Needed to Run this Notebook (20 points)
+
+# In[4]:
+
+
+### DO NOT REMOVE THIS LINE OR EDIT / MOVE THIS CELL ###
+start_time = datetime.now()
+
+
+# The cell below contains the three functions that were used to streamline the amount of code and repitition used. The third function takes the previous two functions, one of which opens a sinlge band and crops it and the other takes the cropped band and masks it using a qa layer, and combines them so that they can be called in the subsequent loop. The loop applies this code to each landsat scene file for both of the sites before outputting a list.
+
+# In[5]:
+
+
+# In this cell place all of the functions needed to run your notebook
+# You will be graded here on function application, docstrings, efficiency so ensure
+# All functions are placed here!
+
+
+#Create paths to both sites
+
+
+
+site_path = os.path.join("ndvi-automation",
+                         "sites")
+
+# path to sjer and harv sites
+study_sites = glob(os.path.join(site_path + "/*/"))
+print(study_sites)
+
+#function to open single landsat band, mask using qa and crop
+def open_clean_bands(band_path,
+                     crop_extent,
+                     valid_range=None,):
+    """Open and mask a single landsat band using a pixel_qa layer.
```

Comment:
This function does not actually use pixel qa layer (?), it opens bands and sets it to valid range. I would amend both doc string and comment above.

Reply:
Thank you for pointing that out. A little slopping from me where I didn't properly edit the docstring when I copied it from another function.